zeebe/broker/src/main/java/io/camunda/zeebe/broker/bootstrap/PartitionManagerStep.java

```diff
@@ -76,5 +93,25 @@ void shutdownInternal(
             shutdownFuture.complete(brokerShutdownContext);
           }
         });
+
+    brokerShutdownContext.getClusterTopology().removeTopologyChangeListener();
+  }
+
+  private void shutdownOnInconsistentTopology(
```

Comment:
I put this here because later we can replace the shutdown with stopping all partitions and restarting based on the new topology.